### DISCUSSION TITLE
about  x-hasura-allowed-roles

### DISCUSSION THREAD

**Discussion by nisharmultani** [hasura/graphql-engine#8872] [Question] [answered]
when i use  x-hasura-allowed-roles: ["editor", "user", "mod"], its work perfect but instead of using direct value i use env that time value goes like that
![v](https://user-images.githubusercontent.com/52249398/186902062-6c3beba1-0b48-4c13-8054-a87c9faea8d0.png)
any solution how to use env in hasura allowed role ?
.

**Comment by adas98012** [comment]
Can you give additional info on what you are setting up in the env var and how you are trying to pass the env var in JWT token ?

**Reply by adas98012** [reply]
Also, here is the doc about how to use JWT token with Hasura GraphQL engine server, for your reference:

https://hasura.io/docs/latest/auth/authentication/jwt/

**Reply by nisharmultani** [reply]
![error](https://user-images.githubusercontent.com/52249398/187355632-ac1ca15a-e95e-4478-a5d8-6f20d3355e6c.png)
when I pass value from env that time in token value goes like below.
![e](https://user-images.githubusercontent.com/52249398/187355873-c0af2b42-e51f-4da5-a61b-b4ff41b4e9e6.png)

**Comment by nisharmultani** [comment, marked as answer]
![2](https://user-images.githubusercontent.com/52249398/187356821-af38d07c-4143-43ac-ae6f-9450160e490b.png)

**Reply by adas98012** [reply]
Can you try using:

"x-hasura-allowed-rules": JSON.parse(process.env.HASURA_ALLOWED_ROLES)

**Comment by nisharmultani** [comment]
yup its work thank you
